lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+    const { container } = rtlRender(),
+        emptyMessage = container.querySelector('.nx-combobox__empty-message');
+
+    expect(emptyMessage).toBeTruthy();
+    expect(emptyMessage).toHaveTextContent('No Results Found');
+  });
+
+  it('sets the empty message from the `emptyMessage` prop', function() {
+    const { container } = rtlRender({ emptyMessage: 'asdfasdf' }),
+        emptyMessage = container.querySelector('.nx-combobox__empty-message');
+
+    expect(emptyMessage).toHaveTextContent('asdfasdf');
+  });
+
+  it('calls onSearch with the current trimmed search text if focus enters the component from elsewhere on the page ' +
+  'while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole} = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox'),
+        anotherElement = document.createElement('button');
+
+    document.body.append(anotherElement);
+    anotherElement.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).toHaveBeenCalledWith('f');
+  });
+
+  it('does not call onSearch if focus moves within the component while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole } = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+
+    const retryBtn = getByRole('button', {name: /retry/i });
+    retryBtn.focus();
+
+    expect(onSearch).not.toHaveBeenCalled();
+    inputElement.focus();
+    expect(onSearch).not.toHaveBeenCalled();
+  });
+
+  it('does not call onSearch if focus moves into the component from an outside window while there is an error',
+      function() {
+        const onSearch = jest.fn(),
+            { getByRole } = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+            inputElement = getByRole('combobox');
+
+        expect(onSearch).not.toHaveBeenCalled();
+        inputElement.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+        (document.activeElement as HTMLElement).blur();
+        expect(onSearch).not.toHaveBeenCalled();
+        inputElement.focus();
+        expect(onSearch).not.toHaveBeenCalled();
+      });
+
+  describe('Keyboard Support', function() {
+    it('sets aria-activeddescendant on the combobox and refers to the focused element id' +
+    'when the element has visual focus', async function() {
+      const { getByRole, getAllByRole } = rtlRender({
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+          inputElement = getByRole('combobox');
+
+      inputElement.focus();
+      await userEvent.keyboard('[ArrowDown]');
+      const optionBtns = getAllByRole('option');
+      expect(inputElement).toHaveAttribute('aria-activedescendant', optionBtns[0].id);
```

Comment:
I'd check successive ArrowDown presses: one when focus is within the list but not on the last item, and then another when focus is on the last item within the list.

Reply:
> one when fake-focus is within the list but not on the last item,

Ah I see this is covered farther down.

Reply:
> when fake-focus is on the last item within the list

I miss this one, here is the updates 8d428e6c8909cbed2c30e6b517581b1f16c97539, thanks.